src/neptune_scale/core/components/sync_process.py

```diff
@@ -461,14 +461,15 @@ def work(self) -> None:
                     logger.debug("Submitting operation #%d with size of %d bytes", sequence_id, len(data))
                     run_operation = RunOperation()
                     run_operation.ParseFromString(data)
-                    request_id = self.submit(operation=run_operation)
+                    request_ids: Optional[SubmitResponse] = self.submit(operation=run_operation)
 
-                    if request_id is None:
+                    if request_ids is None:
                         raise NeptuneUnexpectedError("Server response is empty")
 
-                    logger.debug("Operation #%d submitted as %s", sequence_id, request_id.value)
+                    last_request_id = request_ids.request_ids[-1]
```

Comment:
Check for zero-length response, to error out gracefully?

Reply:
hmm, there's 
```
                    if request_ids is None:
                        raise NeptuneUnexpectedError("Server response is empty")
```

I guess I should check for len(request_ids) == 0 as well